### PR TITLE
update: Re-enable commented code chunks in several notebooks

### PR DESCRIPTION
**Pull request by rmshaffer**
*Issue #, if available:*

*Description of changes:*
Reduces runtime of commented code chunks and ensures that all code is uncommented in several example notebooks:
- **examples/braket_features/Using_the_tensor_network_simulator_TN1.ipynb**: slightly reduce qubit, gate, and shot count to reduce runtime and cost - **new total runtime 2 min (unchanged)**
- **examples/hybrid_jobs/7_Running_notebooks_as_hybrid_jobs/Running_notebooks_as_hybrid_jobs.ipynb**: show cached results from running jobs on multiple devices, explain how to run on QPUs if desired - **new total runtime 4 min (down from 8 min)**
- **examples/pennylane/1_Parallelized_optimization_of_quantum_circuits/1_Parallelized_optimization_of_quantum_circuits.ipynb**: reduce qubit count, reduce warning about long execution time to ~5 min from ~15 min - **new total runtime 0.75 min (unchanged)**
- **examples/pennylane/2_Graph_optimization_with_QAOA/2_Graph_optimization_with_QAOA.ipynb**: simplify problem, use SV1 adjoint gradient (`diff_method="device"`) to speed up by ~10x - **new total runtime 2 min (down from 3 min)**
- **examples/pennylane/4_Simulation_of_noisy_quantum_circuits_on_Amazon_Braket_with_PennyLane/4_Simulation_of_noisy_quantum_circuits_on_Amazon_Braket_with_PennyLane.ipynb**: reduce problem size and iteration count, display cached results rather than including long execution time, re-run notebook to ensure output matches code - **new total runtime 1 min (unchanged)**

Note that despite uncommenting the code, this PR does not increase the runtime of the notebooks (because there are some corresponding reductions in problem size where appropriate).

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by krneta** [comment]
I see a couple of errors in the tests:

> QuantumFunctionError: The braket.aws.qubit device does not provide a native method for computing the jacobian.

Is this happening because you're relying on some PL version?

> ValueError: Result type not found in result. Result types must be added to circuit before circuit is run on device.

This is likely because our mocks are returning generic results and not the type that you expect. This is the [result](https://github.com/amazon-braket/amazon-braket-examples/blob/main/test/integ_tests/pennylane/1_Parallelized_optimization_of_quantum_circuits/1_pennylane_results.json) that we're currently returning for this test (for all calls). Can you check if the result should be modified or new results added?

File: examples/hybrid_jobs/7_Running_notebooks_as_hybrid_jobs/Running_notebooks_as_hybrid_jobs.ipynb [hunk 1]
```diff
@@ -1,400 +1,416 @@
 {
```
Comment (krneta) [inline examples/hybrid_jobs/7_Running_notebooks_as_hybrid_jobs/Running_notebooks_as_hybrid_jobs.ipynb:1]: <p>I'm not sure what we're showcasing here? It seems like the big addition here is the for loop around the AwsQuantumJob.create, and that seems like something we shouldn't need to show here.</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/amazon-braket/amazon-braket-examples/pull/372/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

**Review by krneta — APPROVED** [review]

File: examples/pennylane/2_Graph_optimization_with_QAOA/2_Graph_optimization_with_QAOA.ipynb [hunk 2]
```diff
@@ -22,17 +22,20 @@
    "cell_type": "markdown",
```
Comment (licedric) [inline examples/pennylane/2_Graph_optimization_with_QAOA/2_Graph_optimization_with_QAOA.ipynb:22]: <pre class="ql-syntax" spellcheck="false">Line #3.    n_nodes = 4
</pre><p>Why does the problem need to be shrunk here?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/amazon-braket/amazon-braket-examples/pull/372/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

File: examples/pennylane/4_Simulation_of_noisy_quantum_circuits_on_Amazon_Braket_with_PennyLane/4_Simulation_of_noisy_quantum_circuits_on_Amazon_Braket_with_PennyLane.ipynb [hunk 3]
```diff
@@ -232,7 +232,7 @@
    "outputs": [
```
Comment (licedric) [inline examples/pennylane/4_Simulation_of_noisy_quantum_circuits_on_Amazon_Braket_with_PennyLane/4_Simulation_of_noisy_quantum_circuits_on_Amazon_Braket_with_PennyLane.ipynb:232]: <p>Kind of hard to see convergence here.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/amazon-braket/amazon-braket-examples/pull/372/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='36'/>

File: examples/hybrid_jobs/7_Running_notebooks_as_hybrid_jobs/Running_notebooks_as_hybrid_jobs.ipynb (hunk 1 above)
Comment (rmshaffer) [reply examples/hybrid_jobs/7_Running_notebooks_as_hybrid_jobs/Running_notebooks_as_hybrid_jobs.ipynb:1]: <p>I'll remove the code from this section that runs the multiple jobs - you're right, it's just demonstrating a for loop, there's nothing else new here.</p>

[the remark this answers is not in the document]
Comment (rmshaffer) [reply examples/hybrid_jobs/7_Running_notebooks_as_hybrid_jobs/Running_notebooks_as_hybrid_jobs.ipynb:1]: <p>Good catch, will fix this</p>

File: examples/pennylane/2_Graph_optimization_with_QAOA/2_Graph_optimization_with_QAOA.ipynb (hunk 2 above)
Comment (rmshaffer) [reply examples/pennylane/2_Graph_optimization_with_QAOA/2_Graph_optimization_with_QAOA.ipynb:22]: <p>The optimization run for this problem is done on the LocalSimulator, so shrinking the problem cuts the local runtime roughly in half. Do you feel the demonstration is less valuable/understandable at the smaller size?</p>

File: examples/pennylane/4_Simulation_of_noisy_quantum_circuits_on_Amazon_Braket_with_PennyLane/4_Simulation_of_noisy_quantum_circuits_on_Amazon_Braket_with_PennyLane.ipynb (hunk 3 above)
Comment (rmshaffer) [reply examples/pennylane/4_Simulation_of_noisy_quantum_circuits_on_Amazon_Braket_with_PennyLane/4_Simulation_of_noisy_quantum_circuits_on_Amazon_Braket_with_PennyLane.ipynb:232]: <p>I'll run this out to 25 iterations to make the convergence more obvious on the plot.</p>

File: examples/pennylane/2_Graph_optimization_with_QAOA/2_Graph_optimization_with_QAOA.ipynb (hunk 2 above)
Comment (licedric) [reply examples/pennylane/2_Graph_optimization_with_QAOA/2_Graph_optimization_with_QAOA.ipynb:22]: Ah, surprised that it makes a big enough difference for the local simulator here between 4 and 6 qubits. Are you sure the time savings isn't coming from the fewer number of iterations?

Comment (rmshaffer) [reply examples/pennylane/2_Graph_optimization_with_QAOA/2_Graph_optimization_with_QAOA.ipynb:22]: The number of iterations for this example is still 10 - it hasn't been changed here. The problem size is reduced by shrinking the number of nodes (from 6 to 4) and layers (from 4 to 2).

**Review by krneta — APPROVED** [review]

**Comment by ajberdy** [comment]
Hey, the example repo is frozen for the jobs decorator launch and this PR introduces conflicts with the feature branch (being merged today). I've reverted it in #414 and restored this branch

**Comment by rmshaffer** [comment]
Thanks, and sorry about that. I've copied the changes to a new branch `rmshaffer/uncomment-code-2` and will make a new PR from there after merging and re-running the conflicting notebook.

**Pull request merged**